lib/src/widget/nav_bar/widget/nav_bar_container.dart

```diff
@@ -0,0 +1,159 @@
+import 'package:flutter/material.dart';
+import 'package:xayn_design/src/widget/nav_bar/data/config_pair.dart';
+import 'package:xayn_design/src/widget/nav_bar/widget/nav_bar.dart';
+import 'package:xayn_design/xayn_design.dart';
+
+class NavBarContainer extends StatefulWidget {
+  final MaterialApp child;
+
+  const NavBarContainer({
+    Key? key,
+    required this.child,
+  }) : super(key: key);
+
+  @override
+  _NavBarContainerState createState() => _NavBarContainerState();
+
+  /// Call this, when you need to update current instance of the [NavBar]
+  /// f.e. [NavBarConfig] was changed for the screen
+  static void updateNavBar(BuildContext context) =>
+      context.dependOnInheritedWidgetOfExactType<_Updater>()!.data.updateBar();
+
+  /// This method used internally.
+  /// Will initiate walk through the children widgets,
+  /// find the latest instance of [NavBarConfig] and if it is not [null],
+  /// then will draw [NavBar] with it.
+  /// If not [NavBarConfig] found - [NavBar] will be not shown(removed)
+  ///
+  /// [goingBack] is a flag, that indicates that navigation flow going back.
+  /// when [true], then will ignore latest [NavBarConfig] (from screen that will be removed)
+  /// and will try to apply previous [NavBarConfig] (from screen that will be shown instead)
+  static void resetNavBar(
+    BuildContext context, {
+    required bool goingBack,
+  }) =>
+      context
+          .dependOnInheritedWidgetOfExactType<_Updater>()!
+          .data
+          .resetBar(goingBack);
+}
+
+class _NavBarContainerState extends State<NavBarContainer> {
+  ConfigPair? _configPair;
+
+  Linden get linden => UnterDenLinden.getLinden(context);
+
+  @override
+  Widget build(BuildContext context) {
+    final children = <Widget>[
+      Positioned.fill(child: widget.child),
+    ];
+    final navBarConfig = _configPair?.config;
+    if (navBarConfig != null) {
+      final bar = _buildBar(navBarConfig);
+      children.add(bar);
+    }
+
+    final stack = Stack(children: children);
+
+    final materialApp = MaterialApp(
+      home: stack,
+      theme: widget.child.theme,
+    );
+    return _Updater(
+      data: this,
+      child: materialApp,
+    );
+  }
+
+  /// [goingBack] needed, cos when
+  /// - [Navigator.pop]
+  /// - [Navigator.replace]
+  /// - etc
+  /// are called, then we need to ignore the latest [NavBarConfig],
+  /// which is still part of the widget tree for some time
+  void resetBar(bool goingBack) {
+    WidgetsBinding.instance!.addPostFrameCallback((timeStamp) {
+      if (!mounted) return;
+      final config = _getNavBarConfigPairToShow(context, goingBack);
+      if (config != _configPair) {
+        setState(() {
+          _configPair = config;
+        });
+      }
+    });
+  }
+
+  void updateBar() {
+    WidgetsBinding.instance!.addPostFrameCallback((timeStamp) {
```

Comment:
❓  can you explain in a comment why the `addPostFrameCallback` is needed... (can we avoid it?)

Reply:
I will add also comments to the code 😉 

`updateBar` can be colled during the `build` method. 
Imagine reader mode screen:
- user opens a screen
- we understand that he already liked this article (this understanding comes with new manager state)
- we need to update the navBar widget, but `setState` can not be called during the `build` method

Reply:
hmm but when we call setState during the build method this would cause  another build, wouldn't it?